### ISSUE TITLE
`discardActiveObject` is not prevent moving of object

### ISSUE THREAD

**Issue by vitonsky** [fabricjs/fabric.js#7953]
When i start move object with mouse and call `discardActiveObject`, the active object are reset, but i still move object.
I can't stop moving object programmatically.

## Version
5.0.0

## Test Case
Browser: https://jsfiddle.net/aLuvj3hp/

## Steps To Reproduce

1. Start move red object
2. Wait 3 second, to selection will reset

## Expected Behavior

- Active object are reseted
- Moving of object are stopped

## Actual Behavior

- Active object are reseted
- User still can move object

**Comment by ShaMan123** [comment]
You are right.
We need to address this.
https://jsfiddle.net/yw9t7bu6/1/

**Comment by vitonsky** [comment]
Thank you for temporary solution.

What you think about introduce method like `stopTransformations` instead of change behavior of `discardActiveObject` in new version?

At this time i have not cases when current behavior is necessary, but it's potentially useful.

Or we can add option to `discardActiveObject`.

**Comment by ShaMan123** [comment]
I thought of that as well. But wasn't sure it's useful. Perhaps exposing `stopTransformating` is useful.
I think at this point I'm against adding an option to `discardActiveObject`. It is expected behavior.

**Comment by ShaMan123** [comment]
We can do that and have the dev listen to `deselected` event and call `stopTransformating`.
Perhaps that's more flexible but I wonder why would anyone want to keep transforming an object that isn't active... And is potentially a bug.

**Comment by vitonsky** [comment]
I agree that `discardActiveObject` by default mean "deselect all objects and stop its transforming". This is what i expect when call this method.

About cases when users may need to keep transforming i think about something like this:
- participant selected 3 objects and start move it
- one of its objects been locked by other participant in user application
- user want to implement logic to locked objects will deselect, but want to keep transforming and by release object - rollback transformation

Developer want to keep transforming, to avoid confuse a user.

I think we should save the ability to deselect object, but to don't stop transforming. But we should think how to implement it and update documentation to clarify how it work

**Comment by ShaMan123** [comment]
The case you presented should be handled IMO by adding and removibg objects
to active selection

בתאריך יום ו׳, 20 במאי 2022, 18:05, מאת vitonsky ***@***.***
